Content.Client/Instruments/MidiParser/MidiParser.cs

```diff
@@ -0,0 +1,223 @@
+        "Rhodes Piano", "Chorused Piano", "Harpsichord", "Clavinet",
+        "Celesta", "Glockenspiel", "Music Box", "Vibraphone",
+        "Marimba", "Xylophone", "Tubular Bells", "Dulcimer",
+        "Hammond Organ", "Percussive Organ", "Rock Organ", "Church Organ",
+        "Reed Organ", "Accordion", "Harmonica", "Tango Accordion",
+        "Acoustic Nylon Guitar", "Acoustic Steel Guitar", "Electric Jazz Guitar", "Electric Clean Guitar",
+        "Electric Muted Guitar", "Overdriven Guitar", "Distortion Guitar", "Guitar Harmonics",
+        "Acoustic Bass", "Fingered Electric Bass", "Plucked Electric Bass", "Fretless Bass",
+        "Slap Bass 1", "Slap Bass 2", "Synth Bass 1", "Synth Bass 2",
+        "Violin", "Viola", "Cello", "Contrabass",
+        "Tremolo Strings", "Pizzicato Strings", "Orchestral Harp", "Timpani",
+        "String Ensemble 1", "String Ensemble 2", "Synth Strings 1", "Synth Strings 2",
+        "Choir \"Aah\"", "Voice \"Ooh\"", "Synth Choir", "Orchestra Hit",
+        "Trumpet", "Trombone", "Tuba", "Muted Trumpet",
+        "French Horn", "Brass Section", "Synth Brass 1", "Synth Brass 2",
+        "Soprano Sax", "Alto Sax", "Tenor Sax", "Baritone Sax",
+        "Oboe", "English Horn", "Bassoon", "Clarinet",
+        "Piccolo", "Flute", "Recorder", "Pan Flute",
+        "Bottle Blow", "Shakuhachi", "Whistle", "Ocarina",
+        "Square Wave Lead", "Sawtooth Wave Lead", "Calliope Lead", "Chiff Lead",
+        "Charang Lead", "Voice Lead", "Fiths Lead", "Bass Lead",
+        "New Age Pad", "Warm Pad", "Polysynth Pad", "Choir Pad",
+        "Bowed Pad", "Metallic Pad", "Halo Pad", "Sweep Pad",
+        "Rain Effect", "Soundtrack Effect", "Crystal Effect", "Atmosphere Effect",
+        "Brightness Effect", "Goblins Effect", "Echoes Effect", "Sci-Fi Effect",
+        "Sitar", "Banjo", "Shamisen", "Koto",
+        "Kalimba", "Bagpipe", "Fiddle", "Shanai",
+        "Tinkle Bell", "Agogo", "Steel Drums", "Woodblock",
+        "Taiko Drum", "Melodic Tom", "Synth Drum", "Reverse Cymbal",
+        "Guitar Fret Noise", "Breath Noise", "Seashore", "Bird Tweet",
+        "Telephone Ring", "Helicopter", "Applause", "Gunshot",
+    ];
+
+    // Thanks again to http://www.somascape.org/midi/tech/mfile.html
+    public static bool TryGetMidiTracks(
+        byte[] data,
+        [NotNullWhen(true)] out MidiTrack[]? tracks,
+        [NotNullWhen(false)] out string? error)
+    {
+        tracks = null;
+        error = null;
+
+        var stream = new MidiStreamWrapper(data);
+
+        if (stream.ReadString(4) != "MThd")
+        {
+            error = "Invalid file header";
+            return false;
+        }
+
+        var headerLength = stream.ReadInt32();
+        // MIDI specs define that the header is 6 bytes, we only look at the 6 bytes, if its more, we skip ahead.
+        DebugTools.Assert(headerLength == 6, $"Invalid header length, expected 6, got {headerLength}");
```

Comment:
If you're designing this function to handle invalid formats, you should not throw here.